lib/voorhees/json_api.ex

```diff
@@ -278,4 +278,67 @@ defmodule Voorhees.JSONApi do
   end
 
   defp filter_out_extra_keys(payload, _expected_payload, _options), do: payload
+
+  ## Composable Helpers
+
+  def assert_data(payload, record, opts \\ []) do
+    serialized_record = serialize_record(record, opts)
+
+    data = find_record(payload["data"], serialized_record)
+
+    assert data, "could not find the record with matching id or type in the data"
+
+    Enum.each data["attributes"], fn({key, value}) ->
```

Comment:
I think it would be more helpful to collect the attributes that fail for the resource and display one message, as opposed to error for each attribute